### PR TITLE
test: increase integration coverage on `new/wallet.js`

### PR DESCRIPTION
**Pull request by tuliomir**
This PR increases the coverage of the integration tests on most commonly used cases and updates the current JSDocs for the HathorWallet, so that it is better interpreted by static code analysis.

The integration tests for wallet usage with `xPub` and some edge cases will be implemented on another PR.

### Acceptance Criteria
- Increases the coverage on the `HathorWallet` class
- Updates JSDocs on the `new/wallet.js` and `new/sendTransaction.js` files

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: __tests__/integration/hathorwallet_others.test.js [hunk 1]
```diff
@@ -0,0 +1,1235 @@
...
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: TOKEN_MINT_MASK,
+      }]);
+    expect(hWallet.selectAuthorityUtxo(tokenHash, wallet.isMeltOutput.bind(wallet)))
+      .toStrictEqual([{
+        tx_id: tokenHash,
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: 2,
```
Comment (pedroferreira1) [inline __tests__/integration/hathorwallet_others.test.js:1009]: Why don't you use `TOKEN_MELT_MASK`?

File: __tests__/integration/hathorwallet_others.test.js [hunk 2]
```diff
@@ -0,0 +1,1235 @@
...
+      ]);
+  });
+
+  it('should find many "melt" authority utxos', async () => {
+    // Delegating the mint to another address on the same wallet
+    const meltDelegationTx = await hWallet.delegateAuthority(
+      tokenHash,
+      'melt',
+      hWallet.getAddressAtIndex(1),
+      { createAnother: false }
```
Comment (pedroferreira1) [inline __tests__/integration/hathorwallet_others.test.js:1106]: I would try to do this differently with melt. Instead of finding one spent and one unspent, why don't you create two melt authority utxos and find both in the `hWallet.selectAuthorityUtxo(tokenHash, wallet.isMeltOutput.bind(wallet), { many: true })`? Just to test something new.

File: src/new/wallet.js [hunk 3]
```diff
@@ -768,23 +774,28 @@ class HathorWallet extends EventEmitter {
   }
 
   /**
-   * Get utxos of the wallet addresses
    *
-   * @typedef {Object} UtxoOptions
-   * @property {number} max_utxos - Maximum number of utxos to aggregate. Default to MAX_INPUTS (255).
+   * @typedef UtxoOptions
+   * @property {number} [max_utxos] - Maximum number of utxos to aggregate. Default to MAX_INPUTS (255).
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/new/wallet.js:779]: This code block was attempting to define two types and a function parameter all in the same comment section. The static code analyzers were not able to parse this, so I had to split into 3 different comment sections: the `@typedef` UtxoOptions, `@typedef UtxoDetails` and the function's docstrings.

Looking again at the code I see that the `UtxoOptions.token` parameter is also optional on all places it is used, so I fixed this now.

File: __tests__/integration/hathorwallet_facade.test.js [hunk 4]
```diff
@@ -106,6 +112,27 @@ describe('start', () => {
     }
     hWallet.stop();
   });
+
+  it('should start a multisig wallet', async () => {
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js:116]: We have something similar on the unit tests, but I agree that adding this test also here on the integration would be a good idea.
Reference:
https://github.com/HathorNetwork/hathor-wallet-lib/blob/b6c5a790d3ac3d2ef5e29a9ac330716ff0597d99/__tests__/wallet.test.js#L815-L832

File: src/new/sendTransaction.js [hunk 5]
```diff
@@ -36,8 +36,23 @@ import { OutputType } from '../wallet/types';
  * 'unexpected-error': if an unexpected error happens;
  **/
 class SendTransaction extends EventEmitter {
-  /*
-   * data {Object} Prepared tx data
+  /**
+   *
+   * @param {unknown} [transaction] Full tx data
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/new/sendTransaction.js:41]: I wasn't sure of this, but it's good to make it as specific as possible. Fixed it on [e9856770](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/e9856770773bd22a0d4e09fa17b6b28035aad7df).

File: __tests__/integration/hathorwallet_facade.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js:116]: Implemented on [608f82c3](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/608f82c344e5e16e3ba598499826e59f03281f0a).

File: __tests__/integration/hathorwallet_facade.test.js [hunk 6]
```diff
@@ -699,6 +726,69 @@ describe('sendTransaction', () => {
...
+    // Getting signatures for the proposal
+    const sig1 = mhWallet1.getAllSignatures(txHex, DEFAULT_PIN_CODE);
+    const sig2 = mhWallet2.getAllSignatures(txHex, DEFAULT_PIN_CODE);
+    const sig3 = mhWallet3.getAllSignatures(txHex, DEFAULT_PIN_CODE);
+
+    // Delay to avoid the same timestamp as the fundTx
+    await waitUntilNextTimestamp(mhWallet1, inputTxId);
+
+    // Sign and push
+    const partiallyAsslembledTx = mhWallet1.assemblePartialTransaction(
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js:768]: Fixed on [eb983ff6](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/eb983ff68b73b848ebf78709c4246286f5d7198d)

File: __tests__/integration/hathorwallet_others.test.js (hunk 1 above)
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:1009]: Refactored all uses of these authorities with `1` and `2` to use their respective token masks on [4b175460](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/4b1754600fe2e6bf913b62737ae522ea1be5d8f5).

File: __tests__/integration/hathorwallet_others.test.js (hunk 2 above)
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:1106]: Good idea, implemented this on [f97a9954](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/f97a9954c06758a57da6f1f9bb24a43392690914).

File: __tests__/integration/hathorwallet_facade.test.js [hunk 7]
```diff
@@ -699,6 +726,69 @@ describe('sendTransaction', () => {
     expect(tcba[hWallet.getAddressAtIndex(6)]).toHaveProperty('transactions', 2);
     expect(tcba[hWallet.getAddressAtIndex(12)]).toHaveProperty('transactions', 1);
   });
+
+  it('should send a multisig transaction', async () => {
+    // Initialize multisig wallets and inject funds to test
+    const mhWallet1 = await generateMultisigWalletHelper({ walletIndex: 0 });
+    const mhWallet2 = await generateMultisigWalletHelper({ walletIndex: 1 });
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js:751]: Improved the comments on [e985677](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/e9856770773bd22a0d4e09fa17b6b28035aad7df).

Do you think that is enough for the reader to quickly understand the process?

**Review by pedroferreira1 — APPROVED** [review]

File: __tests__/integration/hathorwallet_others.test.js [hunk 8]
```diff
@@ -0,0 +1,1247 @@
...
+    hWallet2 = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(hWallet2.getAddressAtIndex(0), 101);
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet2,
+      'Consolidate Token',
+      'CTK',
+      100,
+      { address: hWallet2.getAddressAtIndex(0) }
+    );
+    tokenHash = tokenUid;
```
Comment (r4mmer) [inline __tests__/integration/hathorwallet_others.test.js:642]:
```suggestion
    { hash: tokenHash } = await createTokenHelper(
      hWallet2,
      'Consolidate Token',
      'CTK',
      100,
      { address: hWallet2.getAddressAtIndex(0) }
    );
```

File: __tests__/integration/hathorwallet_others.test.js [hunk 9]
```diff
@@ -0,0 +1,1247 @@
...
+
+    // Validation 4
+    expect(hWallet.getUtxosForAmount(10)).toStrictEqual({
+      changeAmount: 0,
+      utxos: [expect.objectContaining({ txId: txHash })]
+    });
+  });
+});
+
+describe('consolidateUtxos', () => {
```
Comment (r4mmer) [inline __tests__/integration/hathorwallet_others.test.js:618]: I think we need 1 more test, trying to consolidate more than the max inputs allowed.
It should consolidate only the max allowed.

File: __tests__/integration/hathorwallet_others.test.js [hunk 10]
```diff
@@ -0,0 +1,1247 @@
...
+    expect(hWallet.selectAuthorityUtxo(
+      HATHOR_TOKEN_CONFIG.uid,
+      () => true,
+      { many: true }
+    )).toStrictEqual([]);
+
+    // Testing the wrapper method
+    expect(hWallet.getAuthorityUtxos(fakeTokenUid, 'mint')).toStrictEqual([]);
+    expect(hWallet.getAuthorityUtxos(fakeTokenUid, 'melt')).toStrictEqual([]);
+    expect(() => hWallet.getAuthorityUtxos(fakeTokenUid, 'invalid')).toThrow();
```
Comment (r4mmer) [inline __tests__/integration/hathorwallet_others.test.js:982]:
```suggestion
    expect(() => hWallet.getAuthorityUtxos(fakeTokenUid, 'invalid')).toThrow('This should never happen.');
```

**Review by r4mmer — CHANGES_REQUESTED** [review]
Splitting the tests in 2 files is great, but I think we should have a `hathorwallet` sub-directory and create the test suites in files like `facade.test.js`, `internals.test.js`, etc.

What do you think?

File: src/new/wallet.js [hunk 11]
```diff
@@ -685,23 +687,27 @@ class HathorWallet extends EventEmitter {
...
-   *
-   * @typedef {Object} AddressInfoOptions
+   * @typedef AddressInfoOptions
    * @property {string} token Optionally filter transactions by this token uid (Default: HTR)
-   *
-   * @typedef {Object} AddressInfo
+   */
+
+  /**
+   * @typedef AddressInfo
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/new/wallet.js:695]: I tried this and got mixed results: VSCode parsed it correctly but Webstorm didn't, so I opted for the syntax that was the most compatible: one declaration per comment block.

I also searched for any JSDocs rule that enforces that only one `typedef` may be implemented per block, but couldn't find any. I just found a nice linting tool that we may use in the future: the [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc).

File: __tests__/integration/hathorwallet_others.test.js (hunk 8 above)
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:642]: Here I first get the `tokenUid` of this transaction from the method results and then I update a variable on the `describe` scope called `tokenHash`.

I tried to make this declaration / assignment more concise, but the best I could think about (aside from what's in the code) was:
```js
const tokenTx = await createTokenHelper(...);
tokenHash = tokenTx.hash;
```

Do you think this would make the code's intentions clearer?

File: __tests__/integration/hathorwallet_facade.test.js [hunk 12]
```diff
@@ -15,16 +15,18 @@ import {
...
 import { loggers } from './utils/logger.util';
 import { SendTxError } from '../../src/errors';
+import SendTransaction from '../../src/new/sendTransaction';
+import helpersUtils from '../../src/utils/helpers';
 
 const fakeTokenUid = '008a19f84f2ae284f19bf3d03386c878ddd15b8b0b604a3a3539aa9d714686e1';
 const sampleNftData = 'ipfs://bafybeiccfclkdtucu6y4yc5cpr6y3yuinr67svmii46v5cfcrkp47ihehy/albums/QXBvbGxvIDEwIE1hZ2F6aW5lIDI3L04=/21716695748_7390815218_o.jpg';
 
-describe('start', () => {
+describe.only('start', () => {
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js:29]: That was incorrect, a leftover from a dev test. Fixed on [1e2156d3](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/1e2156d38f61908d1b24af1b0c63008ae18f0cb2).

File: __tests__/integration/hathorwallet_facade.test.js [hunk 13]
```diff
@@ -106,6 +112,45 @@ describe('start', () => {
     }
     hWallet.stop();
   });
+
+  it.only('should start a multisig wallet', async () => {
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js:116]: That was incorrect, a leftover from a dev test. Fixed on [1e2156d3](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/1e2156d38f61908d1b24af1b0c63008ae18f0cb2).

File: __tests__/integration/hathorwallet_facade.test.js [hunk 14]
```diff
@@ -106,6 +112,45 @@ describe('start', () => {
...
+        pubkeys: multisigWalletsData.pubkeys,
+        numSignatures: 3,
+      },
+    };
+
+    /*
+     * The interaction between the jest infrastructure with the address derivation calculations
+     * somehow make this process very costly and slow, especially for multisig.
+     * Here we lower the gap limit to make this test shorter.
+     */
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js:133]: By using the `preCalculatedAddresses` on the wallet start we skip most of the load that a large `gap_limit` imposes. Only the `start` tests that explicitly avoid this feature, and a single test on `addresses methods`, end up having this extra time by calculating the whole gap limit again.

Most other tests use the addresses on index 0 or 1 to make their assertions, which keep the new address calculations at a minimum. So, I think that lowering the gap limit _for performance only_ would increase complexity for very little gain.

By the way, the next integration test PR 409 will have dedicated tests for the gap limit too :+1:

File: __tests__/integration/hathorwallet_others.test.js [hunk 15]
```diff
@@ -0,0 +1,1247 @@
...
+      })]
+    });
+
+    // Should filter by address
+    expect(hWallet.getUtxosForAmount(10, { filter_address: addr0 })).toStrictEqual({
+      changeAmount: 0,
+      utxos: [expect.anything()]
+    });
+    expect(() => hWallet.getUtxosForAmount(10, { filter_address: addr1 }))
+      .toThrow('utxos to fill total amount');
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:403]: Yes, I preferred to highlight only the relevant part of the message for this specific assertion.

File: __tests__/integration/hathorwallet_others.test.js [hunk 16]
```diff
@@ -0,0 +1,1247 @@
...
+    });
+    expect(hWallet.getAddressInfo(addr1)).toMatchObject({
+      total_amount_received: 10,
+      total_amount_sent: 10,
+      total_amount_available: 0,
+    });
+  });
+
+  it('should throw for an address outside the wallet', async () => {
+    expect(() => hWallet.getAddressInfo(WALLET_CONSTANTS.genesis.addresses[0])).toThrow();
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:83]: Done on [50770592](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/50770592026940601fe1f8dec469bdf98f238376).

File: __tests__/integration/hathorwallet_others.test.js (hunk 10 above)
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:982]: Done on [50770592](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/50770592026940601fe1f8dec469bdf98f238376).

File: src/new/sendTransaction.js [hunk 17]
```diff
@@ -36,8 +36,23 @@ import { OutputType } from '../wallet/types';
...
+   * @param {{
+   *  type: string,
+   *  token: string,
+   *  data?: Buffer,
+   *  address: string,
+   *  value: number,
+   *  timelock: number}[]} outputs
+   * @param {string} [changeAddress]
+   * @param {string} pin
+   * @param {Network} network
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply src/new/sendTransaction.js:55]: Implemented your suggestion in [21c8166b](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/21c8166b9a41340542d4015f4e7543485314a29a) with just one change:

I left the `options` parameter without description and without a default value. This is because some _jsdocs_ interpreters get confused with the parameter initialization and cannot add the children properties later.

File: __tests__/integration/hathorwallet_facade.test.js [hunk 18]
```diff
@@ -1350,13 +1470,13 @@ describe('delegateAuthority', () => {
         txId: duplicateMintAuth,
         index: 0,
         address: hWallet1.getAddressAtIndex(1),
-        authorities: 1 // number value for TOKEN_MINT_MASK
+        authorities: TOKEN_MINT_MASK // number value for TOKEN_MINT_MASK
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js:1473]: Removed on [21c8166b](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/21c8166b9a41340542d4015f4e7543485314a29a).

File: __tests__/integration/hathorwallet_others.test.js [hunk 19]
```diff
@@ -0,0 +1,1247 @@
...
+
+  it('should display correct values for transactions with change', async () => {
+    const addr2 = hWallet.getAddressAtIndex(2);
+    const addr3 = hWallet.getAddressAtIndex(3);
+
+    // Ensure both are empty addresses
+    expect(hWallet.getAddressInfo(addr2).total_amount_received).toStrictEqual(0);
+    expect(hWallet.getAddressInfo(addr3).total_amount_received).toStrictEqual(0);
+
+    // Move all the wallet's funds to "2"
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:94]: Improved on [21c8166b](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/21c8166b9a41340542d4015f4e7543485314a29a). I changed it to the variable name `addr2` itself, which is more in line with other test's comments on the lib.

File: __tests__/integration/hathorwallet_others.test.js [hunk 20]
```diff
@@ -0,0 +1,1247 @@
...
+    const hWallet = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 10);
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet,
+      'getAllUtxos Token',
+      'GAUT',
+      100
+    );
+
+    // Validate that the HTR change is listed
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:303]: Done on [21c8166b](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/21c8166b9a41340542d4015f4e7543485314a29a).

File: __tests__/integration/hathorwallet_others.test.js [hunk 21]
```diff
@@ -0,0 +1,1247 @@
...
+  it('should display correct values for transactions with change', async () => {
+    const addr2 = hWallet.getAddressAtIndex(2);
+    const addr3 = hWallet.getAddressAtIndex(3);
+
+    // Ensure both are empty addresses
+    expect(hWallet.getAddressInfo(addr2).total_amount_received).toStrictEqual(0);
+    expect(hWallet.getAddressInfo(addr3).total_amount_received).toStrictEqual(0);
+
+    // Move all the wallet's funds to "2"
+    let tx = await hWallet.sendTransaction(addr2, 10);
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:95]: Since we know the tests inside a `describe` block will be executed linearly, I declared the `hWallet` HathorWallet on the `describe` block scope. That way and all the `it` blocks reuse the same wallet and inherit the state from the last test.

This was done because the test's scope was restricted to one address at a time, which seemed to allow for a shared wallet. Do you think this made it too confusing?

File: __tests__/integration/hathorwallet_others.test.js [hunk 22]
```diff
@@ -0,0 +1,1247 @@
...
+    expect(hWalletCustom.getAddressInfo(addr0Custom, { token: tokenUid })).toMatchObject({
+      total_amount_received: 100,
+      total_amount_sent: 100,
+      total_amount_available: 0,
+    });
+    expect(hWalletCustom.getAddressInfo(addr1Custom, { token: tokenUid })).toMatchObject({
+      total_amount_received: 40,
+      total_amount_sent: 0,
+      total_amount_available: 40,
+    });
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js]: Added on [fafd503](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/fafd5035e3fe84dcb28b22d7f0cf8956bef187ed).

File: __tests__/integration/hathorwallet_others.test.js [hunk 23]
```diff
@@ -0,0 +1,1247 @@
...
+          locked: false,
+          addressPath: expect.stringMatching(/\/0$/), // Matches "m/44'/280'/0'/0/0", for example
+        }
+      });
+
+    // Expect the generator to have finished
+    expect(await utxoGenerator.next()).toStrictEqual({ value: undefined, done: true });
+  });
+
+  it('should filter by address', async () => {
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:275]: Done on [fafd503](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/fafd5035e3fe84dcb28b22d7f0cf8956bef187ed).

File: __tests__/integration/hathorwallet_others.test.js [hunk 24]
```diff
@@ -0,0 +1,1247 @@
...
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should throw on an empty wallet', async () => {
+    // Should throw for invalid requested amount
+    expect(() => hWallet.getUtxosForAmount(0)).toThrow('positive integer');
+    expect(() => hWallet.getUtxosForAmount(-1)).toThrow('positive integer');
+
+    // Should throw for an amount higher than available funds
+    expect(() => hWallet.getUtxosForAmount(31)).toThrow('utxos to fill total amount');
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js]: I had copied this from another test. On [fafd503](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/fafd5035e3fe84dcb28b22d7f0cf8956bef187ed) changed it to only `1`, as that is sufficient for this scenario.

File: __tests__/integration/hathorwallet_others.test.js [hunk 25]
```diff
@@ -0,0 +1,1247 @@
...
+
+    // Validation 3
+    const utxosGenerator = hWallet.getAllUtxos();
+    expect(utxosGenerator.next()).toStrictEqual({
+      value: undefined,
+      done: true,
+    });
+
+    // Validation 4
+    expect(() => hWallet.getUtxosForAmount(10)).toThrow('utxos to fill');
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:611]: Added on [fafd503](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/fafd5035e3fe84dcb28b22d7f0cf8956bef187ed).

File: __tests__/integration/hathorwallet_others.test.js (hunk 9 above)
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:650]: This test will be implemented on the next PR, together with other validations to the max inputs allowed.

File: __tests__/integration/hathorwallet_others.test.js [hunk 26]
```diff
@@ -0,0 +1,1247 @@
...
+      authorities: TOKEN_MINT_MASK
+    });
+
+    utxoGenResult = await utxoGenerator.next();
+    expect(utxoGenResult.value).toMatchObject({
+      txId: tokenUid,
+      tokenId: tokenUid,
+      value: 2,
+      authorities: TOKEN_MELT_MASK
+    });
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:357]: Good solution to the issue :+1:. Implemented it on [61008dea](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/61008deadb92c9dc346a764b8ab6117ef8fbab9b).

File: __tests__/integration/hathorwallet_others.test.js [hunk 27]
```diff
@@ -0,0 +1,1247 @@
...
+    expect(hWallet.getUtxosForAmount(6, { token: tokenUid, filter_address: addr2 }))
+      .toStrictEqual({
+        changeAmount: 194,
+        utxos: [expect.objectContaining({
+          address: addr2,
+          value: 200,
+        })]
+      });
+    expect(() => hWallet.getUtxosForAmount(6, { token: tokenUid, filter_address: addr3 }))
+      .toThrow('utxos to fill');
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_others.test.js:532]: As this requires changes on the code itself, opened issue to address this:
- #410

File: __tests__/integration/hathorwallet_others.test.js (hunk 8 above)
Comment (r4mmer) [reply __tests__/integration/hathorwallet_others.test.js:671]: My suggestion is that if you use deconstruction to instantiate the local variable, you could use it to update the `tokenHash` variable as well.
But I guess it would make it less clear so I don't think we should change it.

File: __tests__/integration/hathorwallet_others.test.js (hunk 21 above)
[the remark this answers is not in the document]
Comment (r4mmer) [reply __tests__/integration/hathorwallet_others.test.js:97]: I though the wallet was being started with a `beforeEach` which would change the wallet every test.

**Review by r4mmer — APPROVED** [review]
